File: app/src/main/java/com/bitflaker/lucidsourcekit/data/enums/journalratings/DreamClarity.java
package com.bitflaker.lucidsourcekit.data.enums.journalratings;

public enum DreamClarity {
    VeryCloudy("VCL"),
    Cloudy("CLD"),
    Clear("CLR"),
    CrystalClear("CCL"),
    None("");

    private final String id;

    DreamClarity(String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    public static DreamClarity getEnum(String id){
        for (DreamClarity enm : DreamClarity.values()) {
            if (enm.getId().equalsIgnoreCase(id)) {
                return enm;
            }
        }
        return null;
    }
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/main/dreamjournal/rating/DreamClarity.kt
package com.bitflaker.lucidsourcekit.main.dreamjournal.rating

enum class DreamClarity(val id: String) {
    VeryCloudy("VCL"),
    Cloudy("CLD"),
    Clear("CLR"),
    CrystalClear("CCL"),
    None("");

    companion object {
        @JvmStatic
        fun getEnum(id: String) = DreamClarity.entries.find {
            it.id.equals(id, ignoreCase = true)
        }
    }
}
